x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -209,3 +222,66 @@ export const getEndpointAuthzInitialState = (): EndpointAuthz => {
     canReadEventFilters: false,
   };
 };
+
+const SIEM_PERMISSIONS = [
+  { permission: 'canWriteSecuritySolution', privilege: 'crud' },
+  { permission: 'canReadSecuritySolution', privilege: 'show' },
+] as const;
+
+function hasPrivilege(
+  kibanaPrivileges: Array<{
+    resource?: string;
+    privilege: string;
+    authorized: boolean;
+  }>,
+  prefix: string,
+  searchPrivilege: string
+): boolean {
+  const privilege = kibanaPrivileges.find((p) =>
+    p.privilege.endsWith(`${prefix}${searchPrivilege}`)
+  );
+  return privilege?.authorized || false;
+}
+
+export function calculatePermissionsFromPrivileges(
+  kibanaPrivileges:
+    | Array<{
+        resource?: string;
+        privilege: string;
+        authorized: boolean;
+      }>
+    | undefined
+): EndpointPermissions {
+  if (!kibanaPrivileges || !kibanaPrivileges.length) {
+    return defaultEndpointPermissions();
+  }
+
+  const siemPermissions: EndpointPermissions = SIEM_PERMISSIONS.reduce(
+    (acc, { permission, privilege }) => {
```

Comment:
would the siem privilege be set through the kibana privilege toggles as well / are they already available or do we have to add another option underneath security?

Reply:
so these map directly to the top level security all/read in kibana. `siem.crud === security.all` and `siem.show === security.read`. these already existed so I'm just passing them into our `authz` and exposing them as `canReadSecuritySolution` and `canWriteSecuritySolution`.

Reply:
gotcha so as long as this is checked as 'All' it should be ok right? 
<img width="759" alt="Screen Shot 2022-10-10 at 4 45 48 PM" src="https://user-images.githubusercontent.com/56409205/194949904-58543009-c229-4183-90e0-f89ff85f75f4.png">